### ISSUE TITLE
TODO

### ISSUE THREAD

**Issue by garrett** [garrett/cockpit-website-jekyll#1]
The site is coming along, but there are some things that still need to be done. Off the top of my head:

- [x] Fonts
- [x] Author info
- [x] Various style improvements
- [x] Image zoomer (for the front page)
- [x] Use SVGs instead of PNGs (where applicable)
- [x] Gradient on the front page intro
- [x] Mobile tweaks (mainly grids at breakpoints)
- [x] Check URL parity with current site (should already be close as-is)
- [x] Fix offsite links in header nav in the case where the website is in a subdirectory (generally used when testing on GitHub)
- [x] Blog categories (preserve path)
- [x] Blog tags
- [x] Change feed path to the same as on the live site
- [x] Implement category-filtered atom feeds
- [x] Fix 404 header
- [x] Fix search-on-404

**Comment by dperpeet** [comment]
Awesome!
The search functionality is a bit hidden on the landing page... is it worth making this a bit easier to find?

**Comment by petervo** [comment]
Links:

Release Notes and Videos
Developer tutorials

seem to be broken

**Comment by petervo** [comment]
Search page header says:

Document not found
Document not found

**Comment by garrett** [comment]
@dperpeet: I'd like for us to make it more visible in the next iteration. My goal this time around is to port everything over and be feature complete first.

@petervo: What's the URL? I have searching on 404s as well — but I need to fix its duplicate headers. (And it's supposed to look at the requested URL and run a search based on that... but I think that doesn't work right on GitHub at the moment.)

**Comment by petervo** [comment]
@garrett Some document not found examples:

http://garrettlesage.com/cockpit-website-jekyll/blog/category/release.html
http://garrettlesage.com/cockpit-website-jekyll/blog/category/tutorial.html
http://garrettlesage.com/cockpit-website-jekyll/blog/authors/#lars

Also the "Contribute" link in the top nav bar when on a blog or search page is broken.

http://garrettlesage.com/cockpit-website-jekyllhttps://github.com/cockpit-project/cockpit/wiki/Contributing

instead of 

https://github.com/cockpit-project/cockpit/wiki/Contributing

**Comment by garrett** [comment]
@petervo: Thanks!

1. I don't have categories in the right place yet. I'll test those URLs once it's set up.

2. The contribute link is a result of the navigation always putting in the path prefix for links — when it's running locally or on a domain, it's fine. (That's the "Fix offsite links in header nav in the case where the website is in a subdirectory" item.) This wouldn't affect production... but it's a good idea to still fix it for those who fork for staging sites.

3. The blog authors is a bug (that I'm not seeing locally — it's due to using `.html` for some permalinks so we're compatible) — I'll fix that.

**Comment by garrett** [comment]
Currently checking generated file paths of the old site versus the new site and adjusting where necessary...

- I've renamed a few blog posts, as slugs were about 95% correct -- I've handled most, but Jekyll changes dots to dashes in blog slugs and I can't change that (meaning very early versions of Cockpit release blog posts will 404). Everything other than the old releases will now work. (I fixed the 3 with slightly differing slugs.)

- Jekyll cannot auto-create tag or category pages, so tag pages will 404 (I have a single tag page with a tag cloud and it jumps in-page to the grouped categories).

- Since there are only a few categories, I made an include and have stubs for each. The category match for the stub is based on the filename, so one just needs to copy the very simple file and change the filename for a new category. I did the same for filtered RSS feeds.

- Pagination is different; if someone linked to a paginated URL, it will not work. However, things would shift around on Pelican anyway and they would not get what they expect regardless, so it's no big deal, really.

- Images moved around, but I already handled those.

**TL;DR**: All URLs should now work, after comparing paths and making minor adjustments, with the exception of old 0.dot releases and pagination.

**Comment by garrett** [comment]
I'll head to lunch and then work on the search-on-404 feature (which I have working elsewhere) so if someone has an old URL, it'll trigger a search and find the page. (So the old 0.### release blog posts will still point people to the expected place.)

**Comment by garrett** [comment]
Added search on 404 in commit ea2114d.

Examples:

- http://garrettlesage.com/cockpit-website-jekyll/blog/cockpit-0.100.html
- http://garrettlesage.com/cockpit-website-jekyll/blog/cockpit-0.101.html (even works when the title doesn't contain the URL)

Thanks to [the search logic I made (a long while ago)](https://github.com/garrett/cockpit-website-jekyll/blob/f4737e0/assets/lib/search.coffee#L61-L84), the correct pages are the very first hit.

**Comment by garrett** [comment]
Done with all the TODO list!

(I've moved the optional fonts items to their own issue @ #9.)
